src/services/ethereum/CustomSigner.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+
+import { Navigation } from 'react-native-navigation';
+import { screen } from '../../global/Screens';
+
+const ethers = require('ethers');
+
+/**
+ * @desc Custom signer for ethereum RPC
+ * @example
+ * const customSigner = CustomSigner(PRIVATE_KEY, 'homestead')
+ *
+ * @param {string} privateKey Private key of wallet
+ * @param {string} provider name of network
+ * @return {object} custom signer with wallet functions
+ */
+export default function CustomSigner(privateKey: string, provider: string) {
+  const wallet = new ethers.Wallet(privateKey);
+  // wallet.provider = new ethers.providers.InfuraProvider(provider);
+  this.provider = new ethers.providers.InfuraProvider(provider);
+  this.address = wallet.address;
+  this.getBalance = wallet.getBalance;
+  this.estimateGas = wallet.estimateGas;
+  this.getTransactionCount = wallet.getTransaction;
+  this.sign = async (transaction) => {
+    const transactionObject = transaction;
+    try {
+      const signedTransaction = await new Promise((resolve, reject) => {
+        Navigation.showModal({
+          ...screen('CONFIRMATION_SCREEN'),
+          passProps: {
+            onFail: () => {
+              reject();
+            },
+            onSuccess: (gasPrice) => {
+              // Here we have gasPrice to pass it somewhere later.
+              transactionObject.gasPrice = ethers.utils.bigNumberify(`${gasPrice.toString()}000000000`);
+              resolve(wallet.sign(transactionObject));
+            },
+          },
+        });
+      });
+      return signedTransaction;
+    } catch (e) {
+      throw new Error('Transaction aborted!');
+    }
+  };
+  this.sendTransaction = (transaction) => {
+    if (!this.provider) { throw new Error('missing provider'); }
+
+    if (!transaction || typeof (transaction) !== 'object') {
+      throw new Error('invalid transaction object');
+    }
+
+    let { gasLimit } = transaction;
+    if (gasLimit == null) { gasLimit = this.defaultGasLimit; }
+
+    const self = this;
+
+    let gasPricePromise = null;
+    if (transaction.gasPrice) {
+      gasPricePromise = Promise.resolve(transaction.gasPrice);
+    } else {
+      gasPricePromise = this.provider.getGasPrice();
+    }
+
+    let noncePromise = null;
+    if (transaction.nonce) {
+      noncePromise = Promise.resolve(transaction.nonce);
+    } else {
+      noncePromise = this.provider.getTransactionCount(self.address, 'pending');
+    }
+
+    const { chainId } = this.provider;
```

Comment:
I think that the chain it should be chosen based on if it's an development account or not

Reply:
@florianlenz As far as I see it creates a provider based on network name passed [here](https://github.com/Bit-Nation/BITNATION-Pangea-mobile/pull/380/files#diff-64bfab84e2667ab896dcbc653404c041R20), so the provider itself is already based on a network type. Is that what you meant?

Reply:
Yeah it is getting the provider that was passed in when we created the instance of CustomSigner, which is already decided based on dev account